miri-script/src/coverage.rs

```diff
@@ -0,0 +1,84 @@
+use std::ffi::OsString;
+
+use anyhow::{Context, Result};
+use xshell::cmd;
+
+use crate::util::MiriEnv;
+
+/// CoverageReport can generate code coverage reports for miri.
+pub struct CoverageReport {
+
+    /// path is a temporary directory where coverage artifacts will be stored.
+    path: tempfile::TempDir,
+}
+
+impl CoverageReport {
+
+    /// new creates a new CoverageReport
+    /// 
+    /// # Errors
+    /// 
+    /// An error will be returned if a temporary directory could not be created.
+    pub fn new() -> Result<Self> {
+        Ok(Self { path: tempfile::TempDir::new()? })
+    }
+
+    /// add_env_vars will add the required environment variables to MiriEnv `e`.
+    pub fn add_env_vars(&self, e: &mut MiriEnv) -> Result<()> {
+        let rustflags = e.sh.var("RUSTFLAGS")?;
+        let rustflags = format!("{rustflags} -C instrument-coverage");
+        e.sh.set_var("RUSTFLAGS", rustflags);
+
+        let file_template = self.path.path().join("default_%m_%p.profraw");
```

Comment:
Please explain the magic formatting symbols here. Also, why "default"?

Reply:
done. as for "default" i just copy pasted the default value for the env var.

Reply:
Maybe we should just make it `miri_` then to make it a bit more self-explaining :shrug:

Reply:
done.